generators/entity-server/templates/dotnetcore/src/Project.Infrastructure/Data/Repositories/EntityRepository.cs.ejs

```diff
@@ -43,13 +43,13 @@ namespace <%= namespace %>.Infrastructure.Data.Repositories
             <%= camelCasedEntityClass %>.User = null; 
                     <%_ }
             }); _%>
-            if (<%= camelCasedEntityClass %>.Id != 0 && exists) {
-                <%_ relationships.forEach( relationship => {
+            <%_ relationships.forEach( relationship => {
                     if (relationship.relationshipType === 'many-to-many' && relationship.ownerSide) { _%>
-                await RemoveManyToManyRelationship("<%= relationship.joinEntityFieldNamePascalizedPlural %>", "<%= relationship.otherEntityRelationshipNamePascalizedPlural %>Id", "<%= relationship.relationshipFieldNamePascalizedPlural %>Id", <%= camelCasedEntityClass %>.Id, <%= camelCasedEntityClass %>.<%= relationship.relationshipFieldNamePascalizedPlural %>.Select(x => x.Id).ToList());
+            await RemoveManyToManyRelationship("<%= relationship.joinEntityFieldNamePascalizedPlural %>", "<%= relationship.otherEntityRelationshipNamePascalizedPlural %>Id", "<%= relationship.relationshipFieldNamePascalizedPlural %>Id", <%= camelCasedEntityClass %>.Id, <%= camelCasedEntityClass %>.<%= relationship.relationshipFieldNamePascalizedPlural %>.Select(x => x.Id).ToList());
                         <%_ }
-                }); _%>                
-                Update(<%= camelCasedEntityClass %>);
+            }); _%>
+            _context.AddGraph(<%= camelCasedEntityClass %>);
```

Comment:
i have modified here to use AddGraph with create but also with update method because on update with same relation on an entity an error was throw with Update method

Reply:
Ok. 

I have found that working with disconnected entities with EF Core tends to be tricky and unexpected errors can and most likely will happen.

If possible, create an application with entities and relationships set up, and manually creating new entities with the UI and set the relationships. Also update the records using the UI, by un-checking the dropbox/checkbox of the relationship. And check if the changes were properly persisted. 

When issue #397 is implemented in the future, no manual test will be needed, as all this would be unit tesed at the generated application.

Reply:
> 
> 
> Ok.
> 
> I have found that working with disconnected entities with EF Core tends to be tricky and unexpected errors can and most likely will happen.

Yes me too, do you want us to change that ?

> If possible, create an application with entities and relationships set up, and manually creating new entities with the UI and set the relationships. Also update the records using the UI, by un-checking the dropbox/checkbox of the relationship. And check if the changes were properly persisted.
> 

Yes, i have try with https://github.com/jhipster/jhipster-dotnetcore/blob/main/test-integration/samples/jdl-default/app.jdl who don't works today and now all operation is ok 

I will push the blazor sample with relationships after the merge if you want try it